### DISCUSSION TITLE
VueJs Single File Components

### DISCUSSION THREAD

**Discussion by Paracelsus-Rose** [VonHeikemen/nvim-starter#2] [Q&A] [answered]
First off, I just want to thank you for your tireless contributions to the community. Seems I am finding your name around everywhere I look lately, and it is always just top notch stuff. Truly appreciate what you provide.  

The last week and a half or so I have been trying to get lsp / syntax highlighting working for vue.js SFC files.  They are a combination of HTML, JS/TS, and a styling component, (css, sass, scss).  I have tried a few different ways but have not had much success at this time.   I was using lsp-zero, btw amazing work there, finally was able to get things working for rust-tools and tsserver/eslint thanks to that, but now I am attempting to go back to a more simple setup, as to reduce where problems could be happening.  

I know it is beyond the scope of what you are offering here, and you are already providing so much, but if it is at all possible, I would so appreciate some help getting it setup.  Having working vue highlighting / completion would make it so I can finally get back to work and use nvim while I do so.

Once again, thank you so much for all you have done, and if you are unable to help me out on this, I am sure another week of cracking away at it and reading through all the resources you provide, I can hopefully figure it out.  Going to sleep now, will check in later, cheers.

**Comment by VonHeikemen** [comment, marked as answer]
Thank you for the kind words. I'm glad my content helped you.

Okay, so I will assume you want a "vanilla" config, without lsp-zero. 

Syntax highlight with treesitter changed recently, it'll take some time before all colorschemes adapt to this. For now I would recommend pinning `nvim-treesitter` to an old commit. The commit I have in `Plugin-compat.md` seems to work fine. In the list of plugins use this.

```lua
use {'nvim-treesitter/nvim-treesitter', commit = '9bfaf62e42bdcd042df1230e9188487e62a112c0'}
```

Reinstall plugins with the command`:PackerSync`.

Then configure `nvim-treesitter` as usual.

```lua
require('nvim-treesitter.configs').setup({
  highlight = {
    enable = true,
  },
  ensure_installed = {
    'javascript',
    'typescript',
    'css',
    'html',
    'vue'
  },
})
```

For the lsp support you can install `tsserver` `eslint` and `volar`

```sh
npm install -g typescript typescript-language-server 
npm install -g vscode-langservers-extracted
npm install -g @volar/vue-language-server
```

Then you add then in your config.

```lua
local lspconfig = require('lspconfig')

lspconfig.eslint.setup({})
lspconfig.volar.setup({})
lspconfig.tsserver.setup({})
```

**Reply by Paracelsus-Rose** [reply]
You are a god among men.  There are colors finally.  I think it was a combination of fixing one thing, breaking another, all while not realizing the TS symbols were not updated.  Thank you so much for your help.  One day I hope to be able to contribute to others along their journey as you have helped me. Cheers and have a blessed day.
